trade-finance-manager-ui/server/controllers/case/amendments/amendmentBankReviewDate.get.test.js

```diff
@@ -0,0 +1,264 @@
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentBankReviewDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it("should redirect to the 'Has the bank provided a facility end date' page if using FED instead of bank review date", async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE, isUsingFacilityEndDate: true } });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+
+      await getAmendmentBankReviewDate(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+    });
+
+    it('should render the template with isEditable true when the amendment is found and is in progress', async () => {
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE_USING_BANK_REVIEW_DATE });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+      await getAmendmentBankReviewDate(req, res);
+
+      expect(res.render).toHaveBeenCalledWith('case/amendments/amendment-bank-review-date.njk', {
+        dealId,
+        facilityId,
+        dayInput: '',
+        monthInput: '',
+        yearInput: '',
+        isEditable: true,
+        user,
+      });
+    });
+
+    it('should render the template with isEditable false when amendment is found but has been completed', async () => {
+      const COMPLETED_AMENDMENT = { ...MOCK_AMENDMENT_COVERENDDATE_CHANGE_USING_BANK_REVIEW_DATE, status: AMENDMENT_STATUS.COMPLETED };
+
+      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: COMPLETED_AMENDMENT });
+      api.getFacility = jest.fn().mockResolvedValueOnce(gefFacility);
+
+      const req = {
+        params: {
+          _id: dealId,
+          amendmentId,
+          facilityId,
+        },
+        session,
+      };
+      await getAmendmentBankReviewDate(req, res);
+
+      expect(res.render).toHaveBeenCalledWith('case/amendments/amendment-bank-review-date.njk', {
+        dealId,
+        facilityId,
+        dayInput: '',
+        monthInput: '',
+        yearInput: '',
+        isEditable: false,
+        user,
+      });
+    });
+
+    it('should render template prepopulated with the previously entered field values if the bank review date has already been added to the amendment', async () => {
+      const previouslyEnteredBankReviewDate = new Date(2025, 11, 11).toISOString();
```

Comment:
same as the other file, can these be consts so only changed in one place.  Also the it statements are quite long